### PR TITLE
Update/Add security headers

### PR DESCRIPTION
**Pull request by max-debug022**
This PR improves/updates the HTTP-Headers

---
**TLDR**
- Update helmet to 7.2.0 in API and Admin
- Coherent CORS-Policy (Cross-Origin-Ressource-Sharing) for the API -> to Admin & Site
- Coherent CORP/COEP (Cross-Origin-Ressource/Embedder-Policy) for API, Admin & Site
- Strict COOP (Cross-Origin-Opener-Policy) for API, Admin & Site
- Removal of deprecated, redundant, or prohibited headers
- CSP (Content Security Policy) that enforces the JEA (just enough access) principle 
<details>
This PR implements the "Just-Enough-Access" (JEA) security strategy, ensuring that only necessary privileges are granted while restricting all others. The PR includes a coherent CORS policy so that resources can only be shared within the microservices. Additionally, CORP/COEP have been enforced to control resource access and embedding to authorized origins only. Several Content Security Policy (CSP) errors have been resolved & unnecessary privileges have been removed - Admin has slightly more flexibility than the Site. Reasons for several changes:

- Update helmet to v7.1.0
- CORS uses now a maxAge of 600s as Chromium (prior to v76) caps at 10mins 
- CORP only for Same-Site
- Set COOP for same-origin
- Remove the empty exposedHeaders field as it is the default
- Disable unused x-powered-by header
- Disable [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) X-XSS-Protection Header
- Disable [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options) X-Frame-Options Header
- Enable HSTS with recommended values 
- Disable Referrer Policy 
- ...

</details>

### CODE REVIEW

**Comment by thomasdax98** [comment]
Did you test this in one of our projects in a deployed setting? I fear that there might be some unwanted side-effects relating to different domains in a deployed setting

**Comment by max-debug022** [comment]
> Did you test this in one of our projects in a deployed setting? I fear that there might be some unwanted side-effects relating to different domains in a deployed setting

Yes, I tested it on deployed settings.

**Comment by dkarnutsch** [comment]
> Did you test this in one of our projects in a deployed setting? I fear that there might be some unwanted side-effects relating to different domains in a deployed setting

I would not add this to existing projects without extensive testing. However, the starter can define best practices and hopefully all bugs are discovered during development. So I think it is relatively safe to merge this.

**Comment by dkarnutsch** [comment]
Some header could possible have a comment, when to extend/change them.

**Comment by max-debug022** [comment]
> Some header could possible have a comment, when to extend/change them.
> 
> You could also add links to best practices (MDN?), so we can track changes.

Luckily I originally added comments to each Header but removed them since I wasn't sure if comments should be in the code. I could add them, but I think it would be better to add detailed docs on how to modify each header and what effect each header has. What would you prefer?

**Comment by dkarnutsch** [comment]
> > Some header could possible have a comment, when to extend/change them.
> > You could also add links to best practices (MDN?), so we can track changes.
> 
> Luckily I originally added comments to each Header but removed them since I wasn't sure if comments should be in the code. I could add them again, but I think it would be better to add detailed docs on how to modify each header and what effect each header has. What would you prefer?

In code, as documentation can drift away. @johnnyomair?

**Comment by johnnyomair** [comment]
> > > Some header could possible have a comment, when to extend/change them.
> > > You could also add links to best practices (MDN?), so we can track changes.
> > 
> > 
> > Luckily I originally added comments to each Header but removed them since I wasn't sure if comments should be in the code. I could add them again, but I think it would be better to add detailed docs on how to modify each header and what effect each header has. What would you prefer?
> 
> In code, as documentation can drift away. @johnnyomair?

Agreed. We could add line comments for each header/item.

File: site/next.config.js [hunk 1]
```diff
@@ -86,30 +109,11 @@ const nextConfig = {
...
-                            `
-                        .replace(/\s{2,}/g, " ")
-                        .trim(),
+                    value: "same-origin",
                 },
                 {
                     key: "Access-Control-Allow-Origin",
-                    value: process.env.ADMIN_URL,
+                    value: "none",
                 },
```
Comment (johnnyomair) [inline site/next.config.js:117]: Why is it okay to remove this?

File: site/next.config.js [hunk 2]
```diff
@@ -76,6 +89,16 @@ const nextConfig = {
                     key: "Cross-Origin-Opener-Policy",
                     value: "same-origin",
                 },
+                {
+                    key: "Cross-Origin-Embedder-Policy",
+                    // This value should be set to "require-corp" as soon as iframe credentialless it is supported by all browsers
+                    // https://developer.mozilla.org/en-US/docs/Web/Security/IFrame_credentialless
```
Comment (thomasdax98) [inline site/next.config.js:95]:
```suggestion
                    // This value should be set to 'require-corp' as soon as iframe credentialless is supported by all browsers
                    // https://developer.mozilla.org/en-US/docs/Web/Security/IFrame_credentialless
                    // https://caniuse.com/mdn-html_elements_iframe_credentialless
```

File: site/next.config.js (hunk 1 above)
Comment (thomasdax98) [reply site/next.config.js:117]: Is "none" even a valid value? It's not mentioned on MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin

File: site/next.config.js [hunk 3]
```diff
@@ -76,6 +89,16 @@ const nextConfig = {
...
+                    value: "unsafe-none",
+                },
+                {
+                    key: "Cross-Origin-Resource-Policy",
+                    value: "same-site",
```
Comment (thomasdax98) [inline site/next.config.js:100]: Why `same-site` instead of `same-origin`?

File: api/src/main.ts [hunk 4]
```diff
@@ -46,9 +47,34 @@ async function bootstrap(): Promise<void> {
...
+                policy: "no-referrer",
+            },
+            xContentTypeOptions: true,
+            xDnsPrefetchControl: false,
+            xDownloadOptions: true,
+            xPermittedCrossDomainPolicies: true,
+            originAgentCluster: true,
+            crossOriginResourcePolicy: {
+                policy: "same-site",
+            },
```
Comment (thomasdax98) [inline api/src/main.ts:77]: Doesn't this limit the access to the API to our API domain? 

I believe we have some client projects where at least API and site use different domains

File: site/next.config.js (hunk 3 above)
Comment (thomasdax98) [reply site/next.config.js:100]: Ah I guess it's because same-site allows different sub domains

File: site/next.config.js (hunk 1 above)
Comment (max-debug022) [reply site/next.config.js:117]: @johnnyomair  Since this site should only be embedded as an iframe, focusing on CORP/COEP makes sense for security, and it doesn’t need to handle cross-origin requests. Therefore, removing the admin url in Access-Control-Allow-Origin should be ok because CORS mainly hanldes cross-origin data access, typically for APIs.

@thomasdax98 Given that ACAO is unnecessary, we have three options: omit it, set it to an invalid value, or use "none". I prefer setting it to "none" or omitting it entirely. This is because the default behavior for ACAO in Next.js isn’t explicitly documented, and relying on defaults that could change over time isn’t ideal imho.

File: admin/server/server.js [hunk 5]
```diff
@@ -7,24 +7,43 @@ const app = express();
...
+                "font-src": ["'self'", "data:"],
+                "connect-src": ["https:"],
+                "img-src": ["'self'", "data:", "https:"],
+                "frame-src": ["https:"],
+                upgradeInsecureRequests: [],
             },
+            upgradeInsecureRequests: true,
+            useDefaults: false,
         },
         xXssProtection: false,
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply admin/server/server.js:30]: It’s just a personal preference to explicitly set values even if they are implicitly set by default. This way, anyone looking at the code doesn't need to refer to the documentation to understand the defaults. Plus, if the default behavior ever changes, our explicitly set values ensure our application’s behavior remains the same.

File: admin/server/server.js [hunk 6]
```diff
@@ -7,24 +7,43 @@ const app = express();
...
+        xFrameOptions: false,
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply admin/server/server.js:30]: Because it is obsoleted by frame-ancestors: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options

But since frame-ancestors has no fallback, I think we should specify this directive and set it to none or self.

File: admin/server/server.js [hunk 7]
```diff
@@ -7,24 +7,43 @@ const app = express();
...
             maxAge: 63072000,
             includeSubDomains: true,
             preload: true,
         },
+        referrerPolicy: {
+            policy: "no-referrer",
+        },
+        xContentTypeOptions: true,
+        xDnsPrefetchControl: false,
+        xDownloadOptions: true,
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply admin/server/server.js]: Changed in: https://github.com/vivid-planet/comet-starter/pull/280#discussion_r1668052908

File: admin/server/server.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (max-debug022) [reply admin/server/server.js:31]: https://github.com/vivid-planet/comet-starter/pull/280/commits/3221d1ac1eadf4a7e48a8fc85565e258b946234f

File: api/src/app.module.ts [hunk 8]
```diff
@@ -67,7 +67,7 @@ export class AppModule {
                             origin: config.corsAllowedOrigin,
                             methods: ["GET", "POST"],
                             credentials: false,
-                            exposedHeaders: [],
+                            maxAge: 600,
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply api/src/app.module.ts:70]: 1) Reduced Preflight Requests
2) Improved performance by caching preflight requests

File: site/next.config.js [hunk 9]
```diff
@@ -68,6 +68,19 @@ const nextConfig = {
...
             headers: [
+                {
+                    key: "Content-Security-Policy",
+                    value: [
+                        "default-src 'none'",
+                        "script-src-elem 'self'",
+                        "style-src-elem 'unsafe-inline'",
+                        "style-src-attr 'unsafe-inline'",
+                        `img-src 'self' data: ${process.env.API_URL}/`,
+                        "script-src 'unsafe-eval'",
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js:79]: Unfortunately, it seems like it is required by one of our dependencies.

File: site/next.config.js (hunk 3 above)
Comment (max-debug022) [reply site/next.config.js:100]: Subdomains but most importantly ports.

File: api/src/main.ts (hunk 4 above)
Comment (max-debug022) [reply api/src/main.ts:77]: > Doesn't this limit the access to the API to our API domain?
Exactly

> I believe we have some client projects where at least API and site use different domains
We would need to change it in the client project.

**Comment by max-debug022** [comment]
> Agreed. We could add line comments for each header/item.

@johnnyomair what headers should be documented and how? I don't think an explanation is useful since anyone can look headers up and they are fairly well explained. And if you understand the header you probably understand the logic behind their value. 

For example. x-xss-protection - the first thing in the MDN docs is that it should not be used, so its value is false.

File: admin/server/server.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (thomasdax98) [reply admin/server/server.js:30]: > This way, anyone looking at the code doesn't need to refer to the documentation to understand the defaults.

Especially in this case, I'd say it's confusing for someone who isn't very familiar with the headers. XSS protection sounds like a good thing, so people might wonder why we are disabling it. It's not implicit that this header is actually bad. So this will prompt people to consult the docs even more. 

> Plus, if the default behavior ever changes, our explicitly set values ensure our application’s behavior remains the same.

Since this is an outdated, non-standard, harmful header, it's not a realistic scenario that the default ever changes to true. If anything, the header will be removed. While I'm normally also a fan of explicit configuration, I wouldn't say this really applies here.

But let's see what the others think about that

**Comment by thomasdax98** [comment]
> Edit: My initial documentation was somewhat like this:
> 
> ```
> app.use(
>   helmet({
>     contentSecurityPolicy: {
>       directives: {
>         "default-src": ["'none'"], // No external resources are needed
>       },
>       useDefaults: false, // Disable all other default CSP rules
>     },
>     xXssProtection: false, // Disable deprecated header
>     xFrameOptions: false, // Disable deprecated header
>     crossOriginResourcePolicy: "same-origin", // Do not allow cross-origin requests to access the response
>     crossOriginEmbedderPolicy: true, // =no-corp Enable COEP
>     crossOriginOpenerPolicy: true, // =same-origin Enable COOP
>     strictTransportSecurity: {
>       // Enable HSTS
>       maxAge: 63072000, // 2 years (recommended if including subdomains)
>       includeSubDomains: true, // Include all subdomains
>       preload: true, // Enable HSTS preload
>     },
>     referrerPolicy: {
>       policy: "no-referrer", // No referrer information needs to be sent
>     },
>     xContentTypeOptions: true, // = nosniff
>     xDnsPrefetchControl: false, // Disable non-standard header as recommended by MDN
>     xDownloadOptions: true, // = noopen (forces IE8+ not to open files in the context of the page but instead download them)
>     xPermittedCrossDomainPolicies: true, // = none (prevent the browser from MIME-sniffing)
>     originAgentCluster: true, // = false (disables the agent cluster)
>   }),
> );
> ```

@max-debug022 I like this. During the review I had a hard time wrapping my head around "What does `true` (or `false`) mean in this case?"

I think you should add these comments

**Comment by max-debug022** [comment]
I added the comments: https://github.com/vivid-planet/comet-starter/pull/280/commits/cbc047256c8285d648e77fa165b158a0abd119cb

File: admin/server/server.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (max-debug022) [reply admin/server/server.js]: I noticed that false (= not send the header) is not the default value - "0" is. So removed the header since the default value of 0 is actually better than not setting it at all

**Review by thomasdax98 — APPROVED** [review]

File: site/next.config.js (hunk 1 above)
Comment (johnnyomair) [reply site/next.config.js:118]: @max-debug022 but is "none" a valid value?

File: site/next.config.js
Comment (johnnyomair) [inline site/next.config.js:1]: As discussed, please also disable x-powered-by here

**Review by johnnyomair — APPROVED** [review]

**Comment by max-debug022** [comment]
@thomasdax98 @fraxachun @thomasdax98 @johnnyomair

File: admin/server/server.js [hunk 10]
```diff
@@ -7,24 +7,42 @@ const app = express();
...
+                "default-src": ["'none'"],
+                "script-src-elem": ["'unsafe-inline'", "'self'"],
+                "style-src-elem": ["'unsafe-inline'", "'self'"],
+                "style-src-attr": ["'unsafe-inline'"],
+                "font-src": ["'self'", "data:"],
+                "connect-src": ["https:"],
+                "img-src": ["'self'", "data:", "https:"],
+                "frame-src": ["https:"],
+                "frame-ancestors": ["'self'"],
+                upgradeInsecureRequests: [],
```
Comment (dkarnutsch) [inline admin/server/server.js:26]: this is not consistent with the site (where this prop is missing),
also: this prop is not working with our domain setup (dani.dev.vivid-planet.cloud)

Comment (max-debug022) [reply admin/server/server.js:26]: Added with: https://github.com/vivid-planet/comet-starter/pull/280/commits/31542001b14f22273959b9c30dc9db30fe87c471

File: site/next.config.js [hunk 11]
```diff
@@ -24,11 +24,26 @@ const nextConfig = {
...
+                        "connect-src 'self'",
+                        "frame-src https://www.youtube-nocookie.com",
+                        "upgrade-insecure-requests", // Remove this option when working with the Domain-Setup
```
Comment (johnnyomair) [inline site/next.config.js:44]: This should be done automatically based on the presence of the environment variable.

Comment (johnnyomair) [reply site/next.config.js]: I guess you'll need the same fix in the Admin and API, right?

Comment (max-debug022) [reply site/next.config.js]: Yes: https://github.com/vivid-planet/comet-starter/pull/280/commits/7afbbf331528aea4ef10069e4625d7ce4f180fb8

File: site/next.config.js [hunk 12]
```diff
@@ -24,11 +24,25 @@ const nextConfig = {
...
+                    key: "Content-Security-Policy",
+                    value: [
+                        "default-src 'none'",
+                        "script-src-elem 'self' 'unsafe-inline'",
+                        "style-src-elem 'unsafe-inline'",
+                        "style-src-attr 'unsafe-inline'",
+                        `img-src 'self' data: ${process.env.API_URL}/`,
+                        "script-src 'unsafe-eval'",
+                        "connect-src 'self'",
+                        `frame-src https://www.youtube-nocookie.com; ${!process.env.DEV_DOMAIN ? "upgrade-insecure-requests" : ""}`, // Don't use upgrade-insecure-requests with the Domain-Setup
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js]: https://github.com/vivid-planet/comet-starter/pull/280/commits/7afbbf331528aea4ef10069e4625d7ce4f180fb8

**Comment by johnnyomair** [comment]
@max-debug022 please fix conflicts.

File: site/next.config.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js:41]: Well, it's a lot; and I do not know an efficient way to evaluate it (even with tools like bundle-analyzer). Just because a package contains eval or any other "unsafe-eval" code doesn't necessarily mean the CSP needs to permit it. We have hundreds of files (including package dependencies oc) that include some form of eval.

A few example-deps are:
- exceljs
- protobufjs
- next.js
- terser-webpack-plugin
- uglify-js

There are many others, particularly code minifiers, TypeScript dependencies, and linting libraries. Has anyone suggestions on a more effective way to evaluate this?

[the remark this answers is not in the document]
Comment (johnnyomair) [reply site/next.config.js:41]: > Has anyone suggestions on a more effective way to evaluate this?

I'd probably set it to 'none', start the site and see where an error occurs.

> exceljs

I doubt that we have this dependency in the site

[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js:41]: > I'd probably set it to 'none', start the site and see where an error occurs.
That's what I did in the first place, but the error messages are generic and rarely inform you about the specific package.
> I doubt that we have this dependency in the site
Yes, sorry - I forgot to mention that I included all microservices since we not only use the CSP in the site, (although API wouldn't be necessary I guess).

[the remark this answers is not in the document]
Comment (johnnyomair) [reply site/next.config.js:41]: We can have a look at it together the next time we're both in the office.

File: site/next.config.js [hunk 13]
```diff
@@ -24,11 +24,25 @@ const nextConfig = {
...
             source: "/:path*",
             headers: [
+                {
+                    key: "Content-Security-Policy",
+                    value: [
+                        "default-src 'none'",
+                        "script-src-elem 'self' 'unsafe-inline'",
+                        "style-src-elem 'unsafe-inline'",
+                        "style-src-attr 'unsafe-inline'",
+                        `img-src 'self' data: ${process.env.API_URL}/`,
```
Comment (johnnyomair) [inline site/next.config.js:40]: I believe we don't need the API_URL here anymore since we're using a rewrite to proxy the images through the site: #298

**Review by johnnyomair — CHANGES_REQUESTED** [review]
Please fix the lint workflow

**Review by johnnyomair — DISMISSED** [review]

**Comment by johnnyomair** [comment]
@dkarnutsch @fraxachun @thomasdax98 please review, this one's been open for way too long 😁

File: site/next.config.js [hunk 14]
```diff
@@ -24,50 +24,59 @@ const nextConfig = {
...
-                                default-src 'self';
-                                form-action 'self'; 
-                                object-src 'none';
-                                img-src 'self' https: data:${process.env.NODE_ENV === "development" ? " http:" : ""};
-                                media-src 'self' https: data:${process.env.NODE_ENV === "development" ? " http:" : ""};
-                                style-src 'self' 'unsafe-inline'; 
-                                font-src 'self' https: data:;
-                                script-src 'self' 'unsafe-inline' https:${process.env.NODE_ENV === "development" ? " 'unsafe-eval'" : ""};
-                                connect-src 'self' https:${process.env.NODE_ENV === "development" ? " http:" : ""};
-                                frame-ancestors ${process.env.ADMIN_URL};
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js:64]: Yes thats true: https://github.com/vivid-planet/comet-starter/pull/280/commits/e7050c55f36d686e059540befcf4e662c15dab71

File: site/next.config.js [hunk 15]
```diff
@@ -24,50 +24,59 @@ const nextConfig = {
...
+                        "default-src 'self'", // Needed for svgs to work in Firefox (other browsers load svgs with img-src)
+                        "style-src-elem 'self' 'unsafe-inline'",
+                        "style-src-attr 'unsafe-inline'",
+                        "script-src-elem 'self' 'unsafe-inline'",
+                        `${process.env.NODE_ENV === "development" ? "script-src 'unsafe-eval'" : ""}`, // Needed in local development
+                        `${process.env.NODE_ENV === "development" ? "connect-src ws:" : ""}`, // Used for hot reloading in local development
+                        "font-src data:",
+                        "frame-src https://www.youtube-nocookie.com/",
+                        `img-src data: 'self' ${process.env.NODE_ENV === "development" ? process.env.API_URL + "/" : ""}`,
+                        `${process.env.NODE_ENV === "development" ? "" : "upgrade-insecure-requests"}`, // Don't use upgrade-insecure-requests with the Domain-Setup
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js]: What do you think about this: https://github.com/vivid-planet/comet-starter/pull/280/commits/42a5ebfdd8c5ea5aff990f0fb72124c5336a6151

File: site/next.config.js [hunk 16]
```diff
@@ -24,50 +24,59 @@ const nextConfig = {
...
+                    value: [
+                        "default-src 'self'", // Needed for svgs to work in Firefox (other browsers load svgs with img-src)
+                        "style-src-elem 'self' 'unsafe-inline'",
+                        "style-src-attr 'unsafe-inline'",
+                        "script-src-elem 'self' 'unsafe-inline'",
+                        `${process.env.NODE_ENV === "development" ? "script-src 'unsafe-eval'" : ""}`, // Needed in local development
+                        `${process.env.NODE_ENV === "development" ? "connect-src ws:" : ""}`, // Used for hot reloading in local development
+                        "font-src data:",
+                        "frame-src https://www.youtube-nocookie.com/",
+                        `img-src data: 'self' ${process.env.NODE_ENV === "development" ? process.env.API_URL + "/" : ""}`,
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js]: Fixed in: https://github.com/vivid-planet/comet-starter/pull/280/commits/42a5ebfdd8c5ea5aff990f0fb72124c5336a6151

File: site/next.config.js [hunk 17]
```diff
@@ -5,6 +5,31 @@ const withBundleAnalyzer = require("@next/bundle-analyzer")({
     enabled: process.env.ANALYZE === "true",
 });
 
+function generateCSP() {
```
Comment (johnnyomair) [inline site/next.config.js:8]: headers already is a function, so why not move the code there before returning the array?

File: site/next.config.js [hunk 18]
```diff
@@ -5,6 +5,31 @@ const withBundleAnalyzer = require("@next/bundle-analyzer")({
...
+        "img-src": `data: 'self' ${process.env.API_URL}`,
+        "frame-ancestors": process.env.ADMIN_URL,
+    };
+
+    // Conditionally add environment-specific rules
+    if (process.env.NODE_ENV === "development") {
+        cspRules["script-src"] = "'unsafe-eval'"; // Needed in local development
+        cspRules["connect-src"] = "ws:"; // Used for hot reloading in local development
+    } else {
+        cspRules["upgrade-insecure-requests"] = ""; // Don't use upgrade-insecure-requests with Domain-Setup
```
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js:25]: This instructs user agents to treat a site's insecure URLs (HTTP) as though they have been replaced with secure URLs.

**Comment by max-debug022** [comment]
I will wait for this PR to be merged: https://github.com/vivid-planet/comet-starter/pull/415

File: site/next.config.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/next.config.js:25]: Including upgrade-insecure-requests is optional but helps avoid potential misconfigurations and adds an extra layer of security, so I’d recommend keeping it.

**Review by thomasdax98 — DISMISSED** [review]

File: api/package.json [hunk 19]
```diff
@@ -52,8 +52,7 @@
         "compression": "^1.7.5",
         "cookie-parser": "^1.4.7",
         "express": "^4.21.2",
-        "graphql": "^15.10.1",
```
Comment (dkarnutsch) [inline api/package.json:55]: why? this seems unrelated to this PR

Comment (johnnyomair) [reply api/package.json:55]: Also, this shouldn't be removed.

File: api/project-words.txt
Comment (johnnyomair) [inline api/project-words.txt:1]: Please remove this file

File: site/next.config.mjs [hunk 20]
```diff
@@ -25,30 +25,44 @@ const nextConfig = {
...
+                    // https://developer.mozilla.org/en-US/docs/Web/Security/IFrame_credentialless
+                    // https://caniuse.com/mdn-html_elements_iframe_credentialless
+                    value: "unsafe-none",
+                },
+                {
+                    key: "Cross-Origin-Resource-Policy", // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy
+                    value: "same-site", // Do not allow cross-origin requests to access the response
+                },
+                {
+                    //
```
Comment (johnnyomair) [inline site/next.config.mjs:54]:
```suggestion
```

**Review by johnnyomair — CHANGES_REQUESTED** [review]

**Review by johnnyomair — APPROVED** [review]

File: site/src/middleware/contentSecurityPolicyHeaders.ts [hunk 21]
```diff
@@ -2,29 +2,36 @@ import { NextRequest } from "next/server";
...
+type ContentSecurityPolicyDirective = {
+    directive: string;
+    values: string[];
+};
+
+const contentSecurityPolicyDirectives: ContentSecurityPolicyDirective[] = [
+    { directive: "default-src", values: ["'none'"] }, // Restrict any content not explicitly allowed
+    { directive: "style-src-elem", values: ["'self'", "'unsafe-inline'"] },
+    { directive: "script-src-elem", values: ["'self'", "'unsafe-inline'"] },
+    { directive: "img-src", values: ["'self'", "data:", process.env.API_URL?.replace("/api", "") ?? ""] },
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply site/src/middleware/contentSecurityPolicyHeaders.ts:14]: @max-debug022 can we remove the API_URL? The string replacement of /api is odd IMO.

**Comment by johnnyomair** [comment]
> Shouldn't it be comet-demo first?

This PR has been started long before we changed to demo first, so I'd like to finish it here and then copy it to Demo

**Review by dkarnutsch — APPROVED** [review]

File: site/src/middleware/contentSecurityPolicyHeaders.ts (hunk 21 above)
[the remark this answers is not in the document]
Comment (max-debug022) [reply site/src/middleware/contentSecurityPolicyHeaders.ts:14]: We need this for images in the block-preview iframe:
image
image

**Review by thomasdax98 — APPROVED** [review]
